kratos/tests/cpp_tests/geometries/test_line_3d_3.cpp

```diff
@@ -0,0 +1,335 @@
+//
+
+// System includes
+#include <limits>
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "geometries/line_3d_3.h"
+#include "tests/cpp_tests/geometries/test_geometry.h"
+#include "tests/cpp_tests/geometries/test_shape_function_derivatives.h"
+#include "tests/cpp_tests/geometries/cross_check_shape_functions_values.h"
+
+namespace Kratos {
+namespace Testing {
+
+    /// Factory functions
+
+    /** Generates a point type sample Line3D3N
+    * @return  Pointer to a Line3D3N
+    */
+    Line3D3<Point>::Pointer GeneratePointsUnitXDirectionLine3D3() {
+        return Kratos::make_shared<Line3D3<Point>>(
+        Kratos::make_shared<Point>(0.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(1.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(0.5, 0.0, 0.0)
+        );
+    }
+
+    /** Generates a point type sample Line3D3N
+    * @return  Pointer to a Line3D3N
+    */
+    Line3D3<Point>::Pointer GeneratePointsUnitYDirectionLine3D3() {
+        return Kratos::make_shared<Line3D3<Point>>(
+        Kratos::make_shared<Point>(0.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(0.0, 1.0, 0.0),
+        Kratos::make_shared<Point>(0.0, 0.5, 0.0)
+        );
+    }
+
+    /** Generates a point type sample Line3D3N
+    * @return  Pointer to a Line3D3N
+    */
+    Line3D3<Point>::Pointer GeneratePointsDiagonalLine3D3() {
+        return Kratos::make_shared<Line3D3<Point>>(
+        Kratos::make_shared<Point>(0.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(1.0, 1.0, 0.0),
+        Kratos::make_shared<Point>(0.5, 0.5, 0.0)
+        );
+    }
+
+    /** Generates a point type sample Line3D3N
+    * @return  Pointer to a Line3D3N
+    */
+    Line3D3<Point>::Pointer GeneratePointsParabolaLine3D3() {
+        return Kratos::make_shared<Line3D3<Point>>(
+        Kratos::make_shared<Point>(0.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(1.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(0.5, 0.5, 0.0)
+        );
+    }
+
+    /** Generates a point type sample Line3D3N.
+    * @return  Pointer to a Line3D3N
+    */
+    Line3D3<Point>::Pointer GenerateLine3D3WithPoints(Point::Pointer pPointOne, Point::Pointer pPointTwo, Point::Pointer pPointThree ) {
+        return Kratos::make_shared<Line3D3<Point>>(pPointOne, pPointTwo, pPointThree);
+    }
+
+    /** Checks if the number of edges is correct.
+    * Checks if the number of edges is correct.
+    */
+    KRATOS_TEST_CASE_IN_SUITE(Line3D3EdgesNumber, KratosCoreGeometriesFastSuite) {
+        auto p_geometry = GeneratePointsUnitXDirectionLine3D3();
+        KRATOS_CHECK_EQUAL(p_geometry->EdgesNumber(), 2);
+    }
+
+    /** Checks if the number of faces is correct.
+    * Checks if the number of faces is correct.
+    */
+    KRATOS_TEST_CASE_IN_SUITE(Line3D3FacesNumber, KratosCoreGeometriesFastSuite) {
+        auto p_geometry = GeneratePointsUnitXDirectionLine3D3();
+        KRATOS_CHECK_EQUAL(p_geometry->FacesNumber(), 2);
```

Comment:
```suggestion
        KRATOS_CHECK_EQUAL(p_geometry->FacesNumber(), 0);
```

Reply:
Okay, I will unify the quadratic 2d line too